### PR TITLE
qe/prisma-models: delete CompositeType

### PR DESCRIPTION
**Pull request by tomhoule**
Next step in the identifiers saga (https://github.com/prisma/client-planning/issues/265).

### CODE REVIEW

File: query-engine/core/src/query_graph_builder/extractors/filters/composite.rs [hunk 1]
```diff
@@ -19,19 +19,19 @@ pub fn parse(input_map: ParsedInputMap, field: &CompositeFieldRef, _reverse: boo
             (filters::IS_SET, input) => Ok(field.is_set(input.try_into()?)),
 
             // To-many composite filters.
-            (filters::EVERY, input) => Ok(field.every(extract_filter(input.try_into()?, &field.typ())?)),
-            (filters::SOME, input) => Ok(field.some(extract_filter(input.try_into()?, &field.typ())?)),
-            (filters::NONE, input) => Ok(field.none(extract_filter(input.try_into()?, &field.typ())?)),
+            (filters::EVERY, input) => Ok(field.every(extract_filter(input.try_into()?, field.typ())?)),
```
Comment (pimeys) [inline query-engine/core/src/query_graph_builder/extractors/filters/composite.rs:22]: `typ`! bah.

image

File: query-engine/prisma-models/src/parent_container.rs [hunk 2]
```diff
@@ -96,22 +91,16 @@ impl From<ModelWeakRef> for ParentContainer {
...
-    }
-}
-
-impl From<CompositeTypeWeakRef> for ParentContainer {
-    fn from(composite: CompositeTypeWeakRef) -> Self {
         Self::CompositeType(composite)
     }
 }
 
+// impl From<&CompositeTypeRef> for ParentContainer {
```
Comment (pimeys) [inline query-engine/prisma-models/src/parent_container.rs:98]: que?

**Review by pimeys — APPROVED** [review]
yoinks

File: query-engine/core/src/query_graph_builder/extractors/filters/composite.rs (hunk 1 above)
Comment (tomhoule) [reply query-engine/core/src/query_graph_builder/extractors/filters/composite.rs:22]: I always read that as the German "Typ!"

File: query-engine/prisma-models/src/parent_container.rs (hunk 2 above)
Comment (tomhoule) [reply query-engine/prisma-models/src/parent_container.rs:98]: Looks like that can go too :fire:

**Pull request merged**